subsys/settings/src/settings_line.c

```diff
@@ -295,7 +295,15 @@ int settings_line_entry_copy(void *dst_ctx, off_t dst_off, void *src_ctx,
 			break;
 		}
 
-		rc = settings_io_cb.write_cb(dst_ctx, dst_off, buf, chunk_size);
+		size_t write_size = chunk_size;
+
+		if (chunk_size % settings_io_cb.rwbs) {
+			write_size += settings_io_cb.rwbs -
+				      chunk_size % settings_io_cb.rwbs;
+		}
+
+		rc = settings_io_cb.write_cb(dst_ctx, dst_off, buf, write_size);
+
```

Comment:
The only missing part would be changing the line 312 to update `dst_off` with `write_size` instead of `chunk_size`, but since the difference between `chunk_size` and `write_size` occurs only in the last loop run, the current code does not break anything.

Reply:
Actually loop operates on logical data length.